packages/framework/presence/src/test/presenceManager.spec.ts

```diff
@@ -135,157 +195,199 @@ describe("Presence", () => {
 						// Act - simulate join message from client
-						presence.processSignal("", rejoinAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([rejoinAttendeeSignalInfo]);
 
 						// Verify
-						verifyNewAttendee(rejoinAttendeeConnectionId);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
+
+						verifyAttendee(joinedAttendees[0], rejoinAttendeeConnectionId);
 					});
 
 					it.skip('first time is announced via `attendeeJoined` with status "Connected" even if unknown to audience', () => {
 						// Setup - remove connection from audience
 						runtime.removeMember(initialAttendeeConnectionId);
 
 						// Act - simulate join message from client
-						presence.processSignal("", initialAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([initialAttendeeSignalInfo]);
 
 						// Verify
-						verifyNewAttendee(initialAttendeeConnectionId);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
+
+						verifyAttendee(joinedAttendees[0], initialAttendeeConnectionId);
 					});
 
 					it('second time is announced once via `attendeeJoined` with status "Connected" even if most recent unknown to audience', () => {
 						// Setup - remove connection from audience
 						runtime.removeMember(rejoinAttendeeConnectionId);
 
 						// Act - simulate join message from client
-						presence.processSignal("", rejoinAttendeeSignal, false);
+						const joinedAttendees = simulateAttendeeJoin([rejoinAttendeeSignalInfo]);
+						assert(
+							joinedAttendees.length === 1 && joinedAttendees[0] !== undefined,
+							"Expected exactly one attendee to be announced",
+						);
 
-						// Verify
-						verifyNewAttendee(rejoinAttendeeConnectionId);
+						verifyAttendee(joinedAttendees[0], rejoinAttendeeConnectionId);
 					});
 
 					it("as collateral and disconnected is NOT announced via `attendeeJoined`", () => {
 						// Setup - remove connections from audience
 						const collateralAttendeeConnectionId = "client3";
-						const collateralAttendeeSignal = generateBasicClientJoin(clock.now - 10, {
-							averageLatency: 40,
+						runtime.removeMember(initialAttendeeConnectionId);
+						runtime.removeMember(collateralAttendeeConnectionId);
+						const collateralAttendeeSignalInfo = {
 							clientSessionId: attendeeSessionId,
 							clientConnectionId: rejoinAttendeeConnectionId,
 							connectionOrder: 1,
+							fixedTime: clock.now - 10,
 							updateProviders: ["client2"],
 							priorClientToSessionId: {
-								...initialAttendeeSignal.content.data["system:presence"].clientToSessionId,
+								[initialAttendeeConnectionId]: {
+									rev: 0,
+									timestamp: initialAttendeeSignalInfo.fixedTime,
+									value: attendeeSessionId,
+								},
```

Comment:
This is a little unfortunate. Since this is needed at least twice, it should probably be put in a constant next to initialAttendeeSignalInfo and a comment added saying that it is the expected content of signal for initialAttendeeSignalInfo join.
I am assuming there is a need or advantage to have things as `IAttendeeSignalInfo` instead of just the signal. I haven't seen the need scanning through. Maybe in a later change? If not, consider keeping the signal version.

Reply:
Reverted back to signal version, removing IAttendeeSignalInfo. I thought having the helper generate all the signals was 'dry' but specifying the signal info every time results in just as much code.